standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestHiveMetaStore.java

```diff
@@ -3428,4 +3432,44 @@ public void testUpdatePartitionStat_doesNotUpdateStats() throws Exception {
     m.invoke(hms, tbl, part, false, null);
     verify(wh, never()).getFileStatusesForLocation(part.getSd().getLocation());
   }
+
+  @Test(expected = InvalidOperationException.class)
```

Comment:
Won't adding a qtest be better?

You are explicitly doing `tblUpdated.getSd().getBucketCols()` and updating them in the test. qtest will check end to end flow and you don't need to bother about getting Sd and updating the cols explicitly (Just a single alter table will do the trick).